backends/candle/tests/test_modernbert.rs

```diff
@@ -0,0 +1,137 @@
+mod common;
+
+use crate::common::{sort_embeddings, SnapshotEmbeddings};
+use anyhow::Result;
+use common::{batch, cosine_matcher, download_artifacts, load_tokenizer};
+use text_embeddings_backend_candle::CandleBackend;
+use text_embeddings_backend_core::{Backend, ModelType, Pool};
+
+#[test]
+#[serial_test::serial]
+fn test_mini() -> Result<()> {
+    let model_root = download_artifacts("answerdotai/ModernBERT-base", None)?;
+    let tokenizer = load_tokenizer(&model_root)?;
+
+    let backend = CandleBackend::new(
+        &model_root,
+        "float32".to_string(),
+        ModelType::Embedding(Pool::Mean),
+    )?;
+
+    let input_batch = batch(
+        vec![
+            tokenizer.encode("What is Deep Learning?", true).unwrap(),
+            tokenizer.encode("Deep Learning is...", true).unwrap(),
+            tokenizer.encode("What is Deep Learning?", true).unwrap(),
+        ],
+        [0, 1, 2].to_vec(),
+        vec![],
+    );
+
+    let matcher = cosine_matcher();
+
+    let (pooled_embeddings, _) = sort_embeddings(backend.embed(input_batch)?);
+    let embeddings_batch = SnapshotEmbeddings::from(pooled_embeddings);
+    insta::assert_yaml_snapshot!("mini_batch", embeddings_batch, &matcher);
+
+    let input_single = batch(
+        vec![tokenizer.encode("What is Deep Learning?", true).unwrap()],
+        [0].to_vec(),
+        vec![],
+    );
+
+    let (pooled_embeddings, _) = sort_embeddings(backend.embed(input_single)?);
+    let embeddings_single = SnapshotEmbeddings::from(pooled_embeddings);
+
+    insta::assert_yaml_snapshot!("mini_single", embeddings_single, &matcher);
+    assert_eq!(embeddings_batch[0], embeddings_single[0]);
+    assert_eq!(embeddings_batch[2], embeddings_single[0]);
+
+    let input_batch = batch(
+        vec![
+            tokenizer.encode("What is Deep Learning?", true).unwrap(),
+            tokenizer.encode("Deep Learning is...", true).unwrap(),
+        ],
+        [0].to_vec(),
+        [1].to_vec(),
+    );
+
+    let (pooled_embeddings, raw_embeddings) = sort_embeddings(backend.embed(input_batch)?);
+    let pooled_embeddings = SnapshotEmbeddings::from(pooled_embeddings);
+    let raw_embeddings = SnapshotEmbeddings::from(raw_embeddings);
+
+    assert_eq!(embeddings_batch[0], pooled_embeddings[0]);
+    assert_eq!(raw_embeddings.len(), 6);
+
+    Ok(())
+}
+
+#[test]
+#[serial_test::serial]
+fn test_mini_pooled_raw() -> Result<()> {
+    let model_root = download_artifacts("sentence-transformers/all-mpnet-base-v2", None)?;
```

Comment:
shouldn't this be modernbert?

Reply:
thanks for catching this. gonna fix this too

Reply:
fixed https://github.com/huggingface/text-embeddings-inference/pull/459/commits/57a04fce6344e37b1991b8e8152c3670801f6c3d